velox/exec/Aggregate.h

```diff
@@ -427,7 +452,8 @@ AggregateRegistrationResult registerAggregateFunction(
     const std::vector<std::shared_ptr<AggregateFunctionSignature>>& signatures,
     const AggregateFunctionFactory& factory,
     bool registerCompanionFunctions,
-    bool overwrite);
+    bool overwrite,
+    AggregateFunctionMetadata metadata = {});
```

Comment:
It would be nice to move 'metadata' parameter after 'factory' so that all properties that describe the function are together (signatures, factory, metadata).

Maybe add a new method that has 'metadata' argument.

Reply:
Add it to the last can define a default value, if define after 'factory', we can't default a default value, so we have to modify all the place that call registerAggregateFunction.

Should we add a new function like this:
```
AggregateRegistrationResult registerAggregateFunction(
    const std::string& name,
    const std::vector<std::shared_ptr<AggregateFunctionSignature>>& signatures,
    const AggregateFunctionFactory& factory,
    bool registerCompanionFunctions,
    bool overwrite) {
  registerAggregateFunction(name, signatures, factory, AggregateFunctionMetadata, registerCompanionFunctions, overwrite);
}

```

Reply:
@shenh062326 I assume we can keep existing registerAggregateFunction function and add another one with an extra AggregateFunctionMetadata parameter. We can start using the new function for aggregates that are not order-sensitive. Am I missing something?

Reply:
Done.